src/encoderWorker.js

```diff
@@ -329,8 +332,20 @@ OggOpusEncoder.prototype.segmentPacket = function( packetLength ) {
     var segmentLength = Math.min( packetLength, 255 );
     this.segmentTable[ this.segmentTableIndex++ ] = segmentLength;
     var segment = this.encoderOutputBuffer.subarray( packetIndex, packetIndex + segmentLength );
+
+    if ( this.config.streamOpusPackets ) {
+      if ( this.config.framesPerCallback === 1 ) {
+        global['postMessage']({ type: 'opus', data: segment });
+      } else if ( ++this.framesInCallback === this.config.framesPerCallback ) {
+        var cachedSegmentLength = this.segmentTable[ this.segmentTableIndex - 2 ];
+        var cachedSegment = this.segmentData.subarray( this.segmentDataIndex - cachedSegmentLength, this.segmentDataIndex );
+
+        global['postMessage']({ type: 'opus', data: [ cachedSegment, segment ] });
+        this.framesInCallback = 0;
+      }
+    }
+
     this.segmentData.set( segment, this.segmentDataIndex );
```

Comment:
does this modify `segment`?

Reply:
Nope! It just takes the `segment` data and adds it to the `segmentData` array at an offset (`this.segmentDataIndex`)